ext/log_download.py

```diff
@@ -0,0 +1,91 @@
+from datetime import date
+from datetime import timedelta
+import urllib2 as ulib
+import logging
+from BeautifulSoup import BeautifulSoup
+import re
+import time
+import os
+
+root = logging.getLogger()
+root.setLevel(logging.DEBUG)
+
+months = ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sept", "Oct", "Nov", "Dec"]
+UBUNTU_ENDPOINT = "https://irclogs.ubuntu.com/{}/{month:02d}/{day:02d}"
+UBUNTU_CHANNEL_ENDPOINT = "https://irclogs.ubuntu.com/{}/{month:02d}/{day:02d}/%23{channel}"
+SCUMMVM_ENDPOINT = "http://www.enderboi.com/log.php?log=scummvm.log.{day:02d}{month}{year}&format=text"        
+    
+def date_separate(dt):
+    return dt.day, dt.month, dt.year
+
+def save_log(log, save_dir, dt, name):
+    if not (os.path.exists(save_dir) and os.path.isdir(save_dir)):
+        logging.info("Save Directory does exist. Creating a new directory {}".format(save_dir))
+        os.makedirs(save_dir)
+    
+    day,month,year = date_separate(dt)
+    
+    day_dir = os.path.join(save_dir, "{}/{month:02d}/{day:02d}".format(year, month=month, day=day))
+    
+    if not os.path.isdir(day_dir):
+        logging.info("Day log directory does not exist. Creating log directory {}".format(day_dir))
+        os.makedirs(day_dir)
+        
+    with open(os.path.join(day_dir, name), "w+") as f:
+        f.write(log)
+        
+def send_request(url):
+    response = ulib.urlopen(url)
+    
+    status_code = response.getcode()
+    
+    if status_code != 200:
+        logging.warning("Error while sending download request to {} Returned code {}".format(url, status_code))
+    else:
+        logging.info("Log fetched successfully for {}".format(url))
+    
+    res = response.read()
+    time.sleep(1)
+    return res
+    
+
+def ubuntu_url(start_date, end_date):
```

Comment:
a polymorphic method for different IRC channels would be appropriate.

Reply:
@prasadtalasila Sir, the polymorphic method is fetch_logs here.